### ISSUE TITLE
NaN loss issues when adding dropouts layers

### ISSUE THREAD

**Issue by sabeaussan** [state-spaces/mamba#171]
Hi, thank you for this repository. I have been experimenting a bit with the regular model architecture for meta-learning tasks. As such, I tried to add dropouts layers after mamba blocks(right after the call to mamba_inner_fn) to see if it may improves generalization capabilities. The model is training with Pytorch AMP enabled and quickly starts to output a NaN loss. Everything works fine without the added dropout operation. Any idea why this is happening ?

**Comment by apoorv2904** [comment]
Not sure if its the same for you. But for my usecase I found changing precision from bf16 to fp16 resolved the nan issue.

**Comment by HLImg** [comment]
I have the same problem, did you guys solve it?
